### DISCUSSION TITLE
Is FS Notes Sonoma compatible?

### DISCUSSION THREAD

**Discussion by Aetles** [glushchenko/fsnotes#1619] [General]
After upgrading to macOS Sonoma FS Notes crashes on launch (worked fine in Ventura, no changes in files). Is this a known issue?

**Comment by glushchenko** [comment]
Yes, works fine for me. Can you provide crash report?

**Reply by Aetles** [reply]
Yes, here are three of them (I think they all say the same thing). 
[FSNotes crash reports.zip](https://github.com/glushchenko/fsnotes/files/12765238/FSNotes.crash.reports.zip)

I have not yet tried to remove my files from the iCloud folder, but I dit check that all was downloaded locally (no cloud icon).

**Comment by glushchenko** [comment]
This is not related to Sonoma update. Looks like syntax highlighter. Just move last selected note outside configured storage and try again.

**Reply by Aetles** [reply]
Well, that did not work, so I hade to bite the bullet and doing an effort to pinpoint the problematic files. Since I have over 1300 notes it took some time, moving files in and out of iCloud. There were seemingly random crashes so eventually I had to move only a handful at time keeping FSNotes open and when it instantly crashes after a move I moved on to doing it one file at a time. 

After a long time I ended up with 12 files that each caused a crash. The weird thing here is that they are some old notes that hasn't been touched in years. The oldest had a last modified date from 2015! There were both textbundles and text files. So I wondered what has changed? Has iCloud silently corrupted these random files? 

I opened and looked through some of the text files with BBEdit but I could see anything weird or unusual. While poking around FSNotes started to crash again even though all the problematic files were out of there. So I started to pinpoint again and this time it crashed even when all files (text files and textbundles) where removed and only folders left. So I moved to folders out and the crashes stopped. I thought maybe the folders were causing it. But then I added back the folders an no crash. At this point I had no idea of what was going on… 🤷

I removed the folders (it was the standard ones that comes with FSNotes). I consolidated the last text file with images to a textbundle so I could get rid of the i/ folder as well. And then I started to move my notes back and this time there were no crashes at all, even with the problematic files that each and every one could case a crash previously and repeatedly. So no crash at all, and all notes where back (except the folders I removed and all the files in the Trash folder). 

Now I was happy but confused. I remembered that I at some point did turn on the sidebar, to see the folders, and so I hid the sidebar again. And now FSNotes crashes on launch again 😩 

Thankfully I did use Finder tags to mark the 12 crashing files/bundles so I could easily find them again and moved them out of iCloud. And now I could start FSNotes again. 

Now I had a new theory: what if it is the sidebar? 🤔

So I turned on the sidebar again. And now I could add all the problematic files back without crashing FSNotes. And it seemed like each of these files contains hashtags that are shown in the sidebar. So I hid the sidebar and quit FSNotes and started it again: instant crash on launch. 

Moved the files out, re-launched FSNotes, turned on the sidebar, moved the files back in and it worked. As long as the sidebar is not hidden there seems to be no crashes. But as soon as it is hidden, each of these files contain hashtags that crashes when added or on launch. But as long as the sidebar is visible, I have not seen any new crashes, even with all files back in iCloud.

**Reply by glushchenko** [reply]
Please check https://github.com/glushchenko/fsnotes/releases/download/6.4.2/FSNotes_6.4.2.zip

And if you have files that crashed app please send me.

**Reply by Aetles** [reply]
Hi, sorry, I did not have time to test this until now but the newly released version (6.4.3) seems to fix this issue at first glance. I can hide the sidebar with all my files and still launch without crash.
